packages/core/src/models/promptVariants.ts

```diff
@@ -83,6 +83,9 @@ export class PromptVariants {
         slug: {
           equals: slug,
         },
+        variant: {
```

Comment:
This was the one-liner - it was updating all variants to not be current after creating each one. I missed this in testing.